### ISSUE TITLE
[Bug] genearate xml failed when Windows user folder contains when non-ASCII characters

### ISSUE THREAD

**Issue by tsugehara** [firebase/quickstart-unity#1069]
### [REQUIRED] Please fill in the following fields:

 * Unity editor version: 2019.4.9f1, 2019.4.24f1
 * Firebase Unity SDK version: 7.0.2, 7.1.0, 7.2.0
 * Source you installed the SDK: .unitypackage
 * Problematic Firebase Component: FirebaseAnalytics, FirebaseInstanceId,FirebaseRemoteConfig
 * Other Firebase Components in use: NA
 * Additional SDKs you are using: NA
 * Platform you are using the Unity editor on: Windows
 * Platform you are targeting: desktop, Android
 * Scripting Runtime: IL2CPP

### [REQUIRED] Please describe the issue here:

`generate_xml_from_google_services_json.exe` fails when executed by a user whose Windows user folder contains non-ASCII characters.

Example:
```
C:\Users\漢字太郎>c:\Users\Public\unityproject\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe --help
Microsoft.VC90.CRT.manifest could not be extracted!
```
For the exact same project, when the user folder only contains ASCII characters the execution is successful.

Example:
```
C:\Users\taro>c:\Users\Public\unityproject\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe --help
usage: generate_xml_from_google_services_json.exe [-h] [-i FILE] [-o FILE] [-p P] [-l] [-f] [--plist]

Converts a Firebase app/google-services.json into res/values/googleservices.xml similar to the Gradle plugin, or converts a Firebase GoogleService-Info.plist into a
google-services-desktop.json suitible for use on desktop apps.

optional arguments:
  -h, --help  show this help message and exit
  -i FILE     Override input file name
  -o FILE     Override destination file name
  -p P        Package ID to select within the set of packages in the input file. If this is not specified, the first package in the input file is selected.
  -l          List all package IDs referenced by the input file. If this is specified, the output file is not created.
  -f          Print project fields from the input file in the form 'name=value\n' for each field. If this is specified, the output is not created.
  --plist     Specifies a plist file to convert to a JSON configuration file. If this is enabled, the script will expect a .plist file as input, which it will convert
              into google-services-desktop.json file. The output file is *not* suitable for use with Firebase on Android.
```
  
Using pyinstaller I was able to generate an .exe file from `generate_xml_from_google_services_json.py` that could be successfully executed by non-ASCII users. However, the size of the `.exe` increased from 4.9MB to 6.7MB.

Steps to create:

- Python Releases for Windows 3.9.5(amd64)
- pip install pyinstaller
- Copy `Assets/Firebase/Editor/generate_xml_from_google_services_json.py` into any folder
- pyinstaller generate_xml_from_google_services_json.py --onefile

Would it be possible to replace `generate_xml_from_google_services_json.exe` with a version that can be executed by non-ASCII users as well?

### Issues that seem to be related:

- #540: Error loading Python DLL (Generation of the Firebase Android resource file failed)
- #690: Firebase app creation failed on Standalone Windows Build when project contains special charactors
- #1007: [Bug] Crash if windows user has non-ascii name

**Comment by paulinon** [comment]
Hi @tsugehara,

I see that you're facing this issue when your path contains non-ASCII characters. In order to get a better understanding of this behavior, could you verify if the issue persists for each of these conditions:

- The project is clean before adding Firebase.
- You add Firebase as an imported asset without invoking any Firebase functionality.
- You use Firebase the first time by invoking a method.

**Comment by tsugehara** [comment]
@paulb777 Thank you for answering me.

I got this error in clean project.

```
Generation of the Firebase Android resource file google-services.xml from Assets/google-services.json failed.
If you have not included a valid Firebase Android resources in your app it will fail to initialize.

"G:\unity-project\New Unity Project\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe" -i "Assets/google-services.json" -l

Microsoft.VC90.CRT.manifest could not be extracted!



You can start to diagnose this issue by executing ""G:\unity-project\New Unity Project\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe" -i "Assets/google-services.json" -l" from the command line.
UnityEngine.Debug:LogError(Object)
Firebase.Editor.GenerateXmlFromGoogleServicesJson:RunResourceGenerator(IEnumerable`1, String, Boolean) (at Z:/tmp/tmp.OoiDDlawA1/firebase/app/client/unity/editor/src/GenerateXmlFromGoogleServicesJson.cs:542)
Firebase.Editor.GenerateXmlFromGoogleServicesJson:ReadBundleIds(String) (at Z:/tmp/tmp.OoiDDlawA1/firebase/app/client/unity/editor/src/GenerateXmlFromGoogleServicesJson.cs:407)
Firebase.Editor.GenerateXmlFromGoogleServicesJson:UpdateConfigFileDirectory() (at Z:/tmp/tmp.OoiDDlawA1/firebase/app/client/unity/editor/src/GenerateXmlFromGoogleServicesJson.cs:295)
Firebase.Editor.GenerateXmlFromGoogleServicesJson:CheckConfiguration() (at Z:/tmp/tmp.OoiDDlawA1/firebase/app/client/unity/editor/src/GenerateXmlFromGoogleServicesJson.cs:246)
Firebase.Editor.GenerateXmlFromGoogleServicesJson:<GenerateXmlFromGoogleServicesJson>m__1() (at Z:/tmp/tmp.OoiDDlawA1/firebase/app/client/unity/editor/src/GenerateXmlFromGoogleServicesJson.cs:103)
Google.<InitializeOnMainThread>c__AnonStorey0:<>m__0() (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/EditorInitializer.cs:60)
Google.RunOnMainThread:ExecutePollingJobs() (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:409)
Google.RunOnMainThread:<ExecuteAllUnnested>m__1() (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:541)
Google.RunOnMainThread:RunAction(Action) (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:343)
Google.RunOnMainThread:ExecuteAllUnnested(Boolean) (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:530)
Google.RunOnMainThread:ExecuteAll() (at Z:/tmp/tmp.8iabTPBNjq/third_party/unity/unity_jar_resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:512)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
```

I only do this operation.

1. Create New 3D Unity Project (*1)
2. Import Firebase Unity SDKs by Import Packge -> Custom Package of Unity Editor
    - FirebaseInstanceId.unitypackage
    - FirebaseAnalytics.unitypackage
    - FirebaseRemoteConfig.unitypackage
    - Firebase Unity SDK Version: 7.0.2, use dotnet4

(*1) 

![image](https://user-images.githubusercontent.com/3073429/121101741-66e56c00-c837-11eb-986d-935a05942c90.png)

**Comment by chkuang-g** [comment]
@tsugehara

Thank you for reporting this and the detailed workaround.

I suspect that it is due to the executive being generated by Python2. See [this instructions](https://github.com/firebase/firebase-cpp-sdk/blob/main/readme_for_generate_xml_from_google_services_json_exe.txt).

Just to clarify:
The error occurs when you run `generate_xml_from_google_services_json.exe` if the Windows user folder contains non-ASCII characters. It does not matter where you ran it. For instance, if you ran it from `G:\unity-project\` with Windows CommandLine tool, the error still occur when you ran
```
  "G:\unity-project\New Unity Project\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe" -i "Assets/google-services.json" -l
```

Is this accurate?

**Comment by chkuang-g** [comment]
Also, for reference, this looks very similar to #690 
The major difference is: #690 occurs as long as the project name contains non-ASCII character. For your case, it is the Windows user folder, since your project folder `G:\unity-project\New Unity Project\` looks free of non-ASCII characters.

**Comment by chkuang-g** [comment]
Another note: we actually rebuilt `exe` last week using Python 3.
https://github.com/firebase/firebase-cpp-sdk/pull/445

This change is suppose to be included in the next release. But it is still worth it to investigate if that PR actually fixes this issue.

**Comment by tsugehara** [comment]
I tried execute [this file](https://github.com/firebase/firebase-cpp-sdk/blob/89617d469b6b71065b255dd72c8266a2498fe7a8/generate_xml_from_google_services_json.exe) in my windows.

Result:

```
G:\unity-project\New Unity Project\Assets\Firebase\Editor>generate_xml_from_google_services_json.exe -h
usage: generate_xml_from_google_services_json.exe [-h] [-i FILE] [-o FILE] [-p P] [-l] [-f] [--plist]

Converts a Firebase app/google-services.json into res/values/googleservices.xml similar to the Gradle plugin, or
converts a Firebase GoogleService-Info.plist into a google-services-desktop.json suitible for use on desktop apps.

optional arguments:
  -h, --help  show this help message and exit
  -i FILE     Override input file name
  -o FILE     Override destination file name
  -p P        Package ID to select within the set of packages in the input file. If this is not specified, the first
              package in the input file is selected.
  -l          List all package IDs referenced by the input file. If this is specified, the output file is not created.
  -f          Print project fields from the input file in the form 'name=value\n' for each field. If this is
              specified, the output is not created.
  --plist     Specifies a plist file to convert to a JSON configuration file. If this is enabled, the script will
              expect a .plist file as input, which it will convert into google-services-desktop.json file. The output
              file is *not* suitable for use with Firebase on Android.
```

I think this issue has been resolved.

Thank you for supporting the non-ASCII environment, I look forward to the next release of the Firebase SDK.

**Comment by chkuang-g** [comment]
Just to clarify:
Is it resolved because you rebuilt `generate_xml_from_google_services_json.exe`, or is it because you ran it under `G:\unity-project\New Unity Project\Assets\Firebase\Editor`?

**Comment by tsugehara** [comment]
@chkuang-g I haven't rebuilt it, I overwrite local file by [new file](https://github.com/firebase/firebase-cpp-sdk/blob/89617d469b6b71065b255dd72c8266a2498fe7a8/generate_xml_from_google_services_json.exe).

The results of the following two cmmands is the same(old file is error, new file is success):

- `C:\Users\漢字太郎>"g:\unity-project\New Unity Project\Assets\Firebase\Editor\generate_xml_from_google_services_json.exe" -h`
- `G:\unity-project\New Unity Project\Assets\Firebase\Editor>generate_xml_from_google_services_json.exe -h`

**Comment by chkuang-g** [comment]
Oh, that is good to know!

Thank you for the confirmation!

**Comment by chkuang-g** [comment]
This should be fixed in `8.0.0`.  Please let us know if this still occurs.

I will close this for now.

**Comment by tsugehara** [comment]
We have also confirmed that this issue has been fixed. Thank you for your support.
